src/compute/src/compute_state.rs

```diff
@@ -702,40 +697,52 @@ impl PendingPeek {
         })
     }
 
-    fn persist(
+    fn persist<A: Allocate>(
         peek: Peek,
         persist_clients: Arc<PersistClientCache>,
         metadata: CollectionMetadata,
-        active_worker: bool,
+        timely_worker: &mut TimelyWorker<A>,
     ) -> Self {
+        let active_worker = {
+            // Choose the worker that does the actual peek arbitrarily but consistently.
+            let chosen_index = usize::cast_from(peek.uuid.hashed()) % timely_worker.peers();
+            chosen_index == timely_worker.index()
+        };
+
         let (result_tx, result_rx) = oneshot::channel();
         let timestamp = peek.timestamp;
         let mfp_plan = peek.map_filter_project.clone();
         let max_results_needed = peek.finishing.limit.unwrap_or(usize::MAX) + peek.finishing.offset;
 
-        let task_handle = mz_ore::task::spawn(|| "persist::peek", async move {
-            let start = Instant::now();
-            let result = if active_worker {
-                PersistPeek::do_peek(
-                    &persist_clients,
-                    metadata,
-                    timestamp,
-                    mfp_plan,
-                    max_results_needed,
-                )
-                .await
-            } else {
-                Ok(vec![])
-            };
-            let result = match result {
-                Ok(rows) => PeekResponse::Rows(rows),
-                Err(e) => PeekResponse::Error(e.to_string()),
-            };
-            let _ = result_tx.send((result, start.elapsed()));
-        });
+        let button =
+            timely_worker.dataflow_named("Dataflow: persist peek", |scope: &mut Child<_, ()>| {
+                let builder = OperatorBuilder::new("persist peek".to_string(), scope.clone());
+                let token = builder.build(move |capabilities| async move {
+                    let start = Instant::now();
+                    let result = if active_worker {
+                        PersistPeek::do_peek(
```

Comment:
As I mentioned offline, we should do this call in a background task so we're driving futures with tokio instead of timely